dotnet/CoreLib/MemoryStorage/AzureCognitiveSearch/AzureCognitiveSearchMemory.cs

```diff
@@ -419,16 +420,21 @@ private SearchIndex PrepareIndexSchema(string indexName, VectorDbSchema schema)
 
         indexName = this.NormalizeIndexName(indexName);
 
-        const string VectorSearchConfigName = "SemanticMemoryDefaultCosine";
+        const string VectorSearchProfileName = "SemanticMemoryDefaultHnsw";
+        const string VectorSearchHnswConfig = "SemanticMemoryDefaultHnswConfig";
```

Comment:
FYI changed these values to better reflect the new profile/algorithm definitions. Pointing it out in case you have a better naming scheme.

Reply:
have you tested if changing these strings is a breaking change? I think we need to add some documentation answering this questions:

* what if I created an index using different profile name?
* what if I created an index using different name for the search algo?
* why does the search algo have a name? can I use Cosine similarity twice with two different name?
* can I have multiple profiles? what would be the use case?

I suspect there's going to be a lot of "configuration" data to be passed in as soon as we add support for custom indexes.

Since we're going to change these strings now, probably breaking things, could you go with these values for now:

```
const string VectorSearchProfileName = "Default";
const string VectorSearchHnswConfigName = "Default";
```

Reply:
I'll need to do some tests before merging, understanding better these changes and the implications of `Exhaustive`, going through https://learn.microsoft.com/en-us/azure/search/vector-search-how-to-create-index?tabs=config-2023-10-01-Preview%2Crest-2023-10-01-Preview%2Cpush%2Cportal-check-index and trying a combination of old code migrating to new code, old indexes being used by the new code, etc

Reply:
From my (albeit limited) testing of the new API version these are the relevant changes from a SM perspective:

- Vector field profiles. These are defined as part of the index creation. The only thing that can really be configured right now is the nearest-neighbor algorithm used (so cognitive search can index appropriately) but it's probably setting up for future settings. Each vector field needs to reference a profile (previously, they just referenced the algorithm).
- New algorithm, `ExhaustiveKNN`, which (as the name implies) always uses an exhaustive vector search for queries.
- The `Exhaustive` search query flag. If HNSW is used for a field, you can switch to exhaustive KNN during a query, but not vice versa, as the necessary data structures for HNSW won't be available.

To answer your questions (again, my from limited understanding):

- `what if I created an index using different profile name?` No impact, profiles weren't available in prior API versions.
- `what if I created an index using different name for the search algo?` No impact, this is just so cognitive search can setup the index appropriately.
- `why does the search algo have a name? can I use Cosine similarity twice with two different name?` This was so the vector fields could reference an algorithm, same concept as above.
- `can I have multiple profiles? what would be the use case?` Yup, purpose will probably become clearer as the API is built out more, but for right now it's just for using different algorithms.

In short, pre-existing indexes shouldn't be impacted as their structure was already defined (HNSW). The existing default profile/algorithm (HNSW) takes care of the 99% use case imo. Defaulting `Exhaustive` is probably a good idea for now until the ACS methods support a way for the caller to configure the search options a bit more.